frontend/apps/marketing/src/components/common/helpers/index.ts

```diff
@@ -0,0 +1,13 @@
+export const updateUrlParams = (url: string, params: object): string => {
+  const [baseURL, urlQuery = ''] = url.split('?');
+
+  const urlParams = new URLSearchParams(urlQuery);
+  Object.entries(params).forEach(([key, value]) => {
+    urlParams.set(key, value);
+  });
+
+  return `${baseURL}?${urlParams.toString()}`;
```

Comment:
Would this work instead?

```suggestion
  try {
    const url = new URL(url);
    
    url.searchParams.append('fm', 'avif');
    
    return url.toString();
   } catch (e) {
     console.error(e);
     
     return url;
   }
```

Reply:
Unfortunately, the `URL()` constructor cannot directly parse protocol-relative URLs like `//contentful-images.code.org/...`, which we get from Contentful:
<img width="688" alt="Screenshot 2025-06-06 at 00 11 48" src="https://github.com/user-attachments/assets/f274816a-50c7-4b89-b8d9-04ea69dabc81" />

Reply:
I think we'll have to feed it into `getAbsoluteUrl` which will give you the version with https:// in front and then add on fm=avif via `new URL`

Reply:
We do not always get src from entries; it could be a regular string field